Tests/Foundation/Tests/TestFileManager.swift

```diff
@@ -934,6 +934,82 @@ class TestFileManager : XCTestCase {
+
+        try FileManager.default.removeItem(at: baseURL)
+        
+        // A) Check non-symbolic linking resolution
+        try FileManager.default.createDirectory(at: baseURL, withIntermediateDirectories: true)
+        
+        try "test".data(using: .utf8)!.write(to: testFileURL)
+        
+        let resolvedURL_A = testFileURL.resolvingSymlinksInPath().standardized.absoluteURL
+        
+        XCTAssertEqual(resolvedURL_A.path, testFileURL.path)
+        
+        try FileManager.default.removeItem(at: baseURL)
+        
+        // B) Check simple symbolic linking resolution
+        try FileManager.default.createDirectory(at: baseURL, withIntermediateDirectories: true)
+        
+        try "test".data(using: .utf8)!.write(to: testFileURL)
+        
+        try FileManager.default.createSymbolicLink(at: link1URL, withDestinationURL: testFileURL)
+        
+        let resolvedURL_B = link1URL.resolvingSymlinksInPath().standardized.absoluteURL
+        
+        XCTAssertEqual(resolvedURL_B.path, testFileURL.path)
+        
+        try FileManager.default.removeItem(at: baseURL)
+
+        // C) Check recursive symbolic linking resolution
+        try FileManager.default.createDirectory(at: baseURL, withIntermediateDirectories: true)
+
+        try "test".data(using: .utf8)!.write(to: testFileURL)
+        
+        try FileManager.default.createSymbolicLink(at: link1URL, withDestinationURL: link2URL)
+        try FileManager.default.createSymbolicLink(at: link2URL, withDestinationURL: testFileURL)
+        
+        let resolvedURL_C = link1URL.resolvingSymlinksInPath().standardized.absoluteURL
+        XCTAssertEqual(resolvedURL_C.path, testFileURL.path)
+        
+        let resolvedURL_NSString_C = NSString(link1URL.path).resolvingSymlinksInPath.standardizePath()
+        XCTAssertEqual(resolvedURL_NSString_C, testFileURL.path)
+
+        // C-2) And that FileManager.destinationOfSymbolicLink(atPath:) does not recursively resolves them
+        let destinationOfSymbolicLink1 = try FileManager.default.destinationOfSymbolicLink(atPath: link1URL.path)
+        XCTAssertEqual(destinationOfSymbolicLink1, link2URL.path)
+
+        try FileManager.default.removeItem(at: baseURL)
+
+        // D) Check infinite recursion loops are stopped and the function returns the intial symlink
+        try FileManager.default.createDirectory(at: baseURL, withIntermediateDirectories: true)
+        
+        try FileManager.default.createSymbolicLink(at: link1URL, withDestinationURL: link2URL)
+        try FileManager.default.createSymbolicLink(at: link2URL, withDestinationURL: link3URL)
+        try FileManager.default.createSymbolicLink(at: link3URL, withDestinationURL: link1URL)
+
+        let resolvedURL_D = link1URL.resolvingSymlinksInPath()
+        
+        XCTAssertEqual(resolvedURL_D.lastPathComponent, link1URL.lastPathComponent)
+
+        try FileManager.default.removeItem(at: baseURL)
```

Comment:
If some of the `createSymbolicLink()` calls fail this will leave the  test directory lying around after the test has finished.

If you wrap it in 
```
 try withTemporaryDirectory {
    (tmpDirURL, tmpDirPath) in
    var baseURL = tmpDirURL.appendingPathComponent("baseDir")
    .
    .
    .
  }
```

It will create and remove a parent temporary test directory around the whole test.

Reply:
Great, added! Could you start the tests?